File: src/interfaces/notification.ts
import { prop, Typegoose, ModelType, InstanceType } from 'typegoose';

class INotification {

	@prop({ required: true })
	id: number;

	@prop({ required: true })
	created_at: number;

	@prop({ required: true })
	type: string;

	@prop({ required: true })
	content: string;

	@prop()
	account?: string;
}

File: src/interfaces-internal/notification.go
package interfaces_internal

import "github.com/globalsign/mgo/bson"

type INotification struct {
	ID        bson.ObjectId `bson:"_id"`
	CreatedAt int64         `bson:"created_at"`
	Type      string        `bson:"type"`
	Content   string        `bson:"content"`
	Account   string        `bson:"account"`
}
